das/worker.go

```diff
@@ -127,18 +128,25 @@ func (w *worker) sample(ctx context.Context, timeout time.Duration, height uint6
 
 	w.metrics.observeSample(ctx, h, time.Since(start), w.state.jobType, err)
 	if err != nil {
-		if !errors.Is(err, context.Canceled) {
-			log.Debugw(
-				"failed to sample header",
-				"type", w.state.jobType,
-				"height", h.Height(),
-				"hash", h.Hash(),
-				"square width", len(h.DAH.RowRoots),
-				"data root", h.DAH.String(),
-				"err", err,
-				"finished (s)", time.Since(start),
-			)
+		if errors.Is(err, context.Canceled) {
```

Comment:
Also `context.DeadlineExceeded` ?

Reply:
Deadline exceeded would be a failure to retrieve share in given time so it would be a `share ErrNotAvailable`